### PR TITLE
fix(android): fixes errors within lists used for UI

### LINKED ISSUE #3200: bug(android): NestedAdapter creating many UnsupportedOperationExceptions
Sending Log.e to Sentry in #3196 is highlighting this issue in KMEA.

NestedAdapter is an ArrayAdapter that calls addAll() and clear()
From the ArrayAdapter() [documentation](https://developer.android.com/reference/android/widget/ArrayAdapter),  
> underlying data collection being immutable, so methods such as clear() will throw an exception.

This can be triggered when adding a keyboard

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #3200.

Yep.  Turns out that some of NestedAdapter's constructor calls were receiving unmodifiable lists... which, honestly, is a reasonable expectation.  So, the "proper" way forward is to shift to a perspective where the class _expects_ unmodifiable lists and creates its own editable copy.

Also throws in a minor couple of fixes for `build-help.sh`.  A tweak not easily seen by reviewing the code: it lacked the executable bit needed to run on *nix systems.

### CODE REVIEW

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/adapters/NestedAdapter.java [hunk 1]
```diff
@@ -94,10 +94,19 @@ public NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter)
...
-  public NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter, AdapterFilter<Element, A, FilterArg> filter, FilterArg filterArg, List<Element> filteredList) {
+  // In order for the instance to be properly constructed, the `filteredList`
+  // parameter must be mutable.  Unfortunately, there's no way to robustly
+  // check for that, and it illegal to construct a modifiable version
+  // before the super's constructor call b/c Java limitations.
+  // So... we'll keep it locked behind 'protected' for now.
+  //
+  // Could probably throw an extra parameter on for an 'internal' version
+  // and use the current signature as a 'helper' like with the constructor above?
+  protected NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter, AdapterFilter<Element, A, FilterArg> filter, FilterArg filterArg, List<Element> filteredList) {
```
Comment (jahorton) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/adapters/NestedAdapter.java:109]: Fortunately, this fellow's not actually used outside of the class yet.  So, this is sufficient for now.

Java rule:  the `super`-constructor must be called as the first line of a derived class's constructor.  It's literally impossible to in-line wrap the list and save a reference to it here, and `ArrayAdapter` doesn't provide a method to access its internal list later.

File: android/build-help.sh [hunk 2]
```diff
@@ -72,7 +72,11 @@ displayInfo "" \
 cd $KEYMAN_ROOT/android/help
 
 MDLUA="$KEYMAN_ROOT/resources/build/html-link.lua"
-MD=`find -name "*.md"`
+if [[ "${OSTYPE}" == "darwin"* ]]; then
+  MD=`find . -name "*.md"`
+else
+  MD=`find -name "*.md"`
+fi
```
Comment (mcdurdin) [inline android/build-help.sh:79]:
```suggestion
MD=`find . -name "*.md"`
```

I think this should work everywhere

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.260-beta](https://keyman.com/downloads/releases/beta/14.0.260)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.20-alpha](https://keyman.com/downloads/releases/alpha/15.0.20)

**Pull request merged**